Framework/Beamline/src/ComponentInfo.cpp

```diff
@@ -359,5 +507,180 @@ bool ComponentInfo::isStructuredBank(const size_t componentIndex) const {
   return !isDetector(componentIndex) && (*m_isStructuredBank)[rangesIndex];
 }
 
+/**
+ * Get the number of scans for component index
+ * @param index : Component Index
+ * @return Number of scans for component index
+ */
+size_t ComponentInfo::scanCount(const size_t index) const {
+  if (m_detectorInfo && isDetector(index))
+    return m_detectorInfo->scanCount(index);
+  else {
+    return m_scanCounts;
+  }
+}
+
+size_t ComponentInfo::scanSize() const {
+  const auto detectorScanSize = m_detectorInfo ? m_detectorInfo->scanSize() : 0;
+  if (!m_positions)
+    return 0 + detectorScanSize;
+  return m_positions->size() + detectorScanSize;
+}
```

Comment:
I wonder if we also need an `isSyncScan()` method to check whether a component has all sync detector scans. I can't see the `DetectorInfo::isSyncScan()` methods on this branch, perhaps it would be good to merge master?